### PR TITLE
feat[note-settings]: added delete button

### LINKED ISSUE #147: Delete note
Add ability to delete a note from note settings page

### PR DESCRIPTION
**Pull request by e11sy**
- now we can delete the note!
 
image
resolves #147

### CODE REVIEW

File: src/application/services/useNoteSettings.ts [hunk 1]
```diff
@@ -44,6 +45,8 @@ interface UseNoteSettingsComposableState {
    * @param newRole - new role
    */
   changeRole: (id: NoteId, userId: UserId, newRole: MemberRole) => Promise<void>;
+
```
Comment (TatianaFomina) [inline src/application/services/useNoteSettings.ts:48]: missing comment

File: src/domain/noteSettings.repository.interface.ts [hunk 2]
```diff
@@ -41,4 +41,6 @@ export default interface NoteSettingsRepositoryInterface {
    * @param newRole - new role
    */
   patchMemberRoleByUserId(id: NoteId, userId: UserId, newRole: MemberRole): Promise<MemberRole>;
+
+  deleteNote(id: NoteId): Promise<void>;
```
Comment (TatianaFomina) [inline src/domain/noteSettings.repository.interface.ts:45]: missing comment

File: src/presentation/pages/NoteSettings.vue [hunk 3]
```diff
@@ -23,28 +23,33 @@
...
 import type { NoteId } from '@/domain/entities/Note';
 // import TextEdit from '@/presentation/components/form/TextEdit.vue';
-import Button from '@/presentation/components/button/Button.vue';
 import useNoteSettings from '@/application/services/useNoteSettings';
 import { useHead } from 'unhead';
 import { useI18n } from 'vue-i18n';
 import { computed } from 'vue';
-import Team from '@/presentation/components/team/Team.vue';
-import { Section, Row, Switch } from 'codex-ui/vue';
+// import Team from '@/presentation/components/team/Team.vue';
```
Comment (TatianaFomina) [inline src/presentation/pages/NoteSettings.vue:51]: commented out line

File: src/domain/noteSettings.service.ts [hunk 4]
```diff
@@ -99,4 +99,12 @@ export default class NoteService {
   public async patchMemberRoleByUserId(id: NoteId, userId: UserId, newRole: MemberRole): Promise<MemberRole> {
     return await this.noteSettingsRepository.patchMemberRoleByUserId(id, userId, newRole);
   }
+
+  /**
+   *
+   * @param id
```
Comment (elizachi) [inline src/domain/noteSettings.service.ts:105]: Missed description

File: src/infrastructure/noteSettings.repository.ts [hunk 5]
```diff
@@ -65,4 +65,12 @@ export default class NoteSettingsRepository implements NoteSettingsRepositoryInt
   public async patchMemberRoleByUserId(id: NoteId, userId: UserId, newRole: MemberRole): Promise<MemberRole> {
     return await this.transport.patch<MemberRole>(`/note-settings/${id}/team`, { userId, newRole });
   }
+
+  /**
+   *
+   * @param id
```
Comment (elizachi) [inline src/infrastructure/noteSettings.repository.ts:71]: Same problem

File: src/presentation/pages/NoteSettings.vue (hunk 3 above)
Comment (elizachi) [inline src/presentation/pages/NoteSettings.vue:51]: Remove this

**Review by elizachi — APPROVED** [review]

**Review by TatianaFomina — APPROVED** [review]

**Pull request merged**